src/router_core/transfer.c

```diff
@@ -717,6 +725,35 @@ void qdr_delivery_decref_CT(qdr_core_t *core, qdr_delivery_t *dlv, const char *l
         qdr_delete_delivery_internal_CT(core, dlv);
 }
 
+static void qdr_process_tick_CT(qdr_core_t *core, qdr_action_t *action, bool discard)
+{
+    if (discard)
+        return;
+
+    if (DEQ_SIZE(core->timer_list) == 0)
+        return;
```

Comment:
This check is not needed.  If the timer list is empty, the while loop below will exit immediately.